PostQE is now building with CMake, removed the QE download.

enricosiagri opened the pull request:
**PostQE is now building with CMake**
- Now you MUST specify the QE_TOPDIR variable which means you need to have QE downloaded somewhere
- Some files in the build/temp*/postqe-fortran folder have been rearranged.

-  The Makefile responsible for the f90utils module has been renambed f90utils_Makefile
- The Makefile responsibile for the pyqe module has been renamed pyqe_Makefile
- The pyqe_Makefile has been modified to link the libraries and include the modules files generated by CMake
- For now the F90 variable in the pyqe_Makefile has been hardcoded to gfortran (since the following command is using the -E flag)

- The setup.py file has been  modified to use cmake.
- The setupy.py _build_pyqe_module now also search for the FFT, LAPACK and BLAS libraries paths in the CMakeCache.txt file generated by CMake which are then passed when calling cmake on pyqe_Makefile
- The compilation is done using the target qe_pp which probably is compiling more stuff that what is actually needed

**Tested installation process:**
- python 3.8 (with newer python versions there are some errors)
- QE 7.2
- latest ASE, numpy and qeschemas packages
```
pip install -r requirements.txt
QE_TOPDIR=/path/to/your/qe_source_code python setup.py install
```

enricosiagri committed:
PostQE is now building with CMake, removed the QE download.

brunato commented:
I was not able to install using the _gfortran_ compiler. In particular fftw libs were not detected by cmake. After installing `fftw-devel` package the fftw libs has been detected and the installation works.

brunato commented:
With Python 3.11 running the tests produce this error:

```
$ python tests/test_postqe.py 
Traceback (most recent call last):
  File "/home/brunato/Development/postqe/tests/test_postqe.py", line 19, in <module>
    from postqe import EspressoCalculator, PostqeCalculator, get_band_structure
  File "/home/brunato/.virtualenvs/postqe-dev/lib/python3.11/site-packages/postqe/__init__.py", line 11, in <module>
    from .calculator import EspressoCalculator, PostqeCalculator
  File "/home/brunato/.virtualenvs/postqe-dev/lib/python3.11/site-packages/postqe/calculator.py", line 11, in <module>
    from modulefinder import EXTENDED_ARG
ImportError: cannot import name 'EXTENDED_ARG' from 'modulefinder' (/usr/lib64/python3.11/modulefinder.py)
```

Trying with Python 3.8 works but many tests fails.

enricosiagri commented:
Yes. It works with Python 3.8 for now. The tests needs to be fixed as I
think they were failing also with the previous version where configure and
make was used.

On Tue, 23 May 2023, 18:51 Davide Brunato, ***@***.***> wrote:

> With Python 3.11 running the tests produce this error:
>
> $ python tests/test_postqe.py
> Traceback (most recent call last):
>   File "/home/brunato/Development/postqe/tests/test_postqe.py", line 19, in <module>
>     from postqe import EspressoCalculator, PostqeCalculator, get_band_structure
>   File "/home/brunato/.virtualenvs/postqe-dev/lib/python3.11/site-packages/postqe/__init__.py", line 11, in <module>
>     from .calculator import EspressoCalculator, PostqeCalculator
>   File "/home/brunato/.virtualenvs/postqe-dev/lib/python3.11/site-packages/postqe/calculator.py", line 11, in <module>
>     from modulefinder import EXTENDED_ARG
> ImportError: cannot import name 'EXTENDED_ARG' from 'modulefinder' (/usr/lib64/python3.11/modulefinder.py)
>
> Trying with Python 3.8 works but many tests fails.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QEF/postqe/pull/5#issuecomment-1559816575>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOIEA2MT74I5CNVBSWE4GEDXHTTIZANCNFSM6AAAAAAYLZUPTQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

enricosiagri committed:
commented EXTENDED_ARG as it was imported but never used and a minor tweak in the setup.py file

enricosiagri commented:
> from modulefinder import EXTENDED_ARG
> ImportError: cannot import name 'EXTENDED_ARG' from 'modulefinder' (/usr/lib64/python3.11/modulefinder.py)

I commented out the import of EXTENDED_ARGS as it was imported but never used. Now it shoudl work with python 3.11

enricosiagri commented:
> I was not able to install using the _gfortran_ compiler. In particular fftw libs were not detected by cmake. After installing `fftw-devel` package the fftw libs has been detected and the installation works.

Yes, I think, this  might be a problem related to how I parsed the CMakeCache.txt file. I assumed that the LIBRARY PATHS where always displayed in the same way. We will fix it.

brunato commented:
> > from modulefinder import EXTENDED_ARG
> > ImportError: cannot import name 'EXTENDED_ARG' from 'modulefinder' (/usr/lib64/python3.11/modulefinder.py)
> 
> I commented out the import of EXTENDED_ARGS as it was imported but never used. Now it shoudl work with python 3.11

Maybe remove instead, if it's not used it's useless ...

brunato commented on postqe/fortran/wrapfiles/funct.f90:
Why add QE source f90 files to postqe repo?

pietrodelugas replied:
I am not sure that file comes directly from QE source, it was probably changed and adapted

brunato reviewed (commented):
Maybe some changes are needed.

pietrodelugas committed:
adapt pyqe build setup to multistring libraries

e.g. mkl

pietrodelugas committed:
several changes:

1. refactoring get_eigenvalues for efficiency:
   * all eigenvalues are read from xml and chached in an array
   * get_eigevalues retrieves a slice of the cache array

2. If schema is not provided in input the calculator looks for the
   data-file-schema.xml file and fetches the schema indicated therein

pietrodelugas committed:
for nspin 2 the two doses are plotted

pietrodelugas committed:
fixing band structure methods

forgotten things about calculator

pietrodelugas committed:
updating example 2

pietrodelugas committed:
fix cli emin emax

pietrodelugas committed:
updating example3

pietrodelugas committed:
wrappers cropped to only definition and documentation
